parser/prism/Translator.cc

```diff
@@ -254,22 +258,11 @@ std::unique_ptr<parser::Node> Translator::translate(pm_node_t *node) {
         case PM_CONSTANT_PATH_NODE: {
             // Part of a constant path, like the `A` in `A::B`. `B` is a `PM_CONSTANT_READ_NODE`
             auto constantPathNode = reinterpret_cast<pm_constant_path_node *>(node);
-            pm_location_t *loc = &constantPathNode->base.location;
-
-            std::string_view name = parser.resolveConstant(constantPathNode->name);
-
-            std::unique_ptr<parser::Node> parent;
-            if (constantPathNode->parent) {
-                // This constant reference is chained onto another constant reference.
-                // E.g. if `node` is pointing to `B`, then then `A` is the `parent` in `A::B::C`.
-                parent = translate(constantPathNode->parent);
-            } else { // This is a fully qualified constant reference, like `::A`.
-                pm_location_t *delimiterLoc = &constantPathNode->delimiter_loc; // The location of the `::`
-                parent = make_unique<parser::Cbase>(parser.translateLocation(delimiterLoc));
-            }
-
-            return make_unique<parser::Const>(parser.translateLocation(loc), std::move(parent),
-                                              gs.enterNameConstant(name));
```

Comment:
I have pulled this out into a helper method because depending on the context, Sorbet has different representations of `PM_CONSTANT_PATH_NODE`s. If it's a path node on its own, then Sorbet represents it as a `Const`, but if it's being assigned a value, it's a `ConstLhs`.